bin/pre-install.js

```diff
@@ -0,0 +1,99 @@
+// The "userAgent" is something like 'yarn/0.28.1 npm/? node/v8.4.0 linux x64'
+// when ran with yarn. With npm it's similar, without the `yarn` part of course.
+const userAgent = process.env.npm_config_user_agent;
+if (!userAgent) {
+  console.error('Error: this script cannot be run directly.');
+  process.exit(-1);
+}
+
+const agents = userAgent.split(' ').reduce((agents, agent) => {
+  const [key, value] = agent.split('/');
+  agents[key] = value;
+  return agents;
+}, {});
+
+const checks = [checkNode(), checkYarn()];
+
+const successful = checks.every(returnValue => returnValue);
+if (successful) {
+  console.log(
+    'All project requirements are satisfied, moving forward with the installation.'
+  );
+  process.exit(0);
+}
+process.exit(-1);
+
+function checkNode() {
+  // Node versions usually have a starting `v`.
+  const strNodeVersion = agents.node.replace(/^v/, '');
+  // parseFloat will get us major.minor and ignore the possible patch version, which is
+  // enough for our needs because node versions very rarely (or never ?) have
+  // patch versions.
+  const nodeVersion = parseFloat(strNodeVersion);
```

Comment:
This does not work for node `v7.10.1`, parseFloat turns this into `7.1` which is less than `7.8`

Reply:
Oh, such a beginner mistake. I'm ashamed.